features/cellular/framework/targets/QUECTEL/EC2X/QUECTEL_EC2X.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2019, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "QUECTEL_EC2X.h"
+
+#include "PinNames.h"
+#include "AT_CellularNetwork.h"
+#include "rtos/ThisThread.h"
+
+using namespace mbed;
+using namespace rtos;
+using namespace events;
+
+static const intptr_t cellular_properties[AT_CellularBase::PROPERTY_MAX] = {
+    AT_CellularNetwork::RegistrationModeLAC,    // C_EREG
+    AT_CellularNetwork::RegistrationModeLAC,    // C_GREG
+    AT_CellularNetwork::RegistrationModeEnable,    // C_REG
+    1,  // AT_CGSN_WITH_TYPE
```

Comment:
Should be 0?

Reply:
Why ? AT+CGSN command is supported by the modem. Am I missing something here ?

Reply:
Yes, it supports the command but without <snt> parameter, see 3GPP TS 27.0.0.7 chapter 5.4.